pkg/apis/core/helper/qos/qos.go

```diff
@@ -39,7 +39,7 @@ func GetPodQOS(pod *core.Pod) core.PodQOSClass {
 	limits := core.ResourceList{}
 	zeroQuantity := resource.MustParse("0")
 	isGuaranteed := true
-	for _, container := range pod.Spec.Containers {
+	for _, container := range append(pod.Spec.Containers, pod.Spec.InitContainers...) {
```

Comment:
appending to a slice we don't own can end up sharing memory in unpleasant ways. prefer double iterating or appending to a slice we allocated (didn't check how frequently this is called to know which is better)

Reply:
ok, changed pushed